core/sdk-core/src/it/java/software/amazon/awssdk/core/http/ConnectionPoolMaxConnectionsIntegrationTest.java

```diff
@@ -79,7 +82,7 @@ public void leasing_a_new_connection_fails_with_connection_pool_timeout() {
                       .request(request)
                       .originalRequest(NoopTestRequest.builder().build())
                       .executionContext(executionContext(request))
-                      .execute();
+                      .execute(combinedSyncResponseHandler(null, null));
```

Comment:
Same as previous question

Reply:
The execute() method on the client was internal and was only ever called by tests. I consider that an antipattern so I removed it, and made the test do a bit of extra heavy lifting instead.

Reply:
The null is handled by combinedSyncResponseHandler() which is a test helper method. There's nothing in the actual code that expects or handles a null for a handler (although there used to be through that test-only path).